### ISSUE TITLE
Implement logging for moderation events

### ISSUE THREAD

**Issue by dhruvkb** [WordPress/openverse#3943]
## Description

To send metrics from the Django API to AWS CloudWatch where they will be visualised, the metrics will be sent as structured logs that can be parsed by AWS Logs Insights.

Add logging for moderation events when `Report` and `Decision` models are created/modified. The logging should be emitted from `save` method on these models.

## Additional context

- https://github.com/WordPress/openverse/issues/3432
- [Implementation plan](https://docs.openverse.org/projects/proposals/trust_and_safety/content_report_moderation/20240122-implementation_plan_moderation_metrics.html#implement-real-time-metrics)

**Comment by dhruvkb** [comment]
The addition of logging to ModerationDecision is tightly coupled with #3636 and can be implemented simultaneously or after that issue is resolved.

**Comment by dhruvkb** [comment]
This is blocked on #3432.

**Comment by sarayourfriend** [comment]
Reprioritised as per https://github.com/WordPress/openverse/issues/383#issuecomment-2151103330

**Comment by sarayourfriend** [comment]
Closing as won't do as per https://github.com/WordPress/openverse/issues/383#issuecomment-2153354643
